### ISSUE TITLE
Usernames with Null Byte Characters accepted in registration page

### ISSUE THREAD

**Issue by NullArgument** [Hackademic/hackademic#125]
If someone registers with the null byte character "%00" or with "%0a", "%0b", "%0c", ...

![screen1](https://cloud.githubusercontent.com/assets/8127545/5602718/caf391fe-9369-11e4-9261-ab9e2e2c25b7.png)

The administrator will get this in the user listing:

![screen2](https://cloud.githubusercontent.com/assets/8127545/5602719/d1c5c72c-9369-11e4-9e85-0314ec411795.png)

Also, if someone registers with "admin%00", it will look like he's the admin to others.
Basically, he can impersonate any username.

![screen3](https://cloud.githubusercontent.com/assets/8127545/5602727/659c2072-936a-11e4-9384-bb54e4b3b448.png)

**Comment by pchaigno** [comment]
Good catch! Are you working on a fix?

**Comment by NullArgument** [comment]
@pchaigno 
I will when i'm done with the fuzzing of the application.

**Comment by NullArgument** [comment]
@pchaigno Something else, is it really ok to accept the same email multiple times during registration?

**Comment by pchaigno** [comment]
Nop, it shouldn't be possible :/

**Comment by NullArgument** [comment]
@pchaigno 
Then it must be fixed as well.

**Comment by NullArgument** [comment]
Fixed in ed971467c46ab63fd732e64a1872f73ea6c5f643
